examples/jsm/renderers/common/Backend.js

```diff
@@ -114,15 +114,18 @@ class Backend {
 
 	getDomElement() {
 
-		let domElement = this.domElement;
+		if ( this.domElement === null ) {
 
-		if ( domElement === null ) {
+			const { canvas = this.createCanvasElement() } = this.parameters;
 
-			this.domElement = domElement = ( this.parameters.canvas !== undefined ) ? this.parameters.canvas : this.createCanvasElement();
+			// OffscreenCanvas does not have setAttribute, see #22811
+			if ( 'setAttribute' in canvas ) canvas.setAttribute( 'data-engine', `three.js r${REVISION}` );
+
+			this.domElement = canvas;
```

Comment:
Can we keep the original code and just and the following line inside the `if` statement?
```js
if ( 'setAttribute' in domElement ) domElement.setAttribute( 'data-engine', `three.js r${REVISION}` );
```